mdpow/workflows/dihedrals.py

```diff
@@ -323,30 +322,40 @@ def dihedral_groups_ensemble(dirname, atom_group_indices,
 
     return df
 
-def save_df(df, df_save_dir=None, resname=None, molname=None):
+def save_df(df, df_save_dir, resname=None, molname=None):
     '''Takes a :class:`pandas.DataFrame` of results from 
        :class:`~mdpow.analysis.dihedral.DihedralAnalysis`
        as input before padding the angles to optionaly save the raw
-       data before plotting dihedral angle frequencies as KDE violins
+       data.
+
+       Optionally saves results before padding the angles for periodicity
+       and plotting dihedral angle frequencies as KDE violins
        with :func:`~mdpow.workflows.dihedrals.dihedral_violins`.
-       Given a parent directory, creates subdirectory
-       for molecule, saves fully sampled csv.
+       Given a parent directory, creates subdirectory for molecule,
+       saves fully sampled, unpadded results as compressed csv.
 
        :keywords:
 
-       *df*
-           results :class:`pandas.DataFrame` from
-           :class:`~mdpow.analysis.dihedral.DihedralAnalysis`
+        *df*
+           :class:`pandas.DataFrame` of :class:`~mdpow.analysis.dihedral.DihedralAnalysis`
+           results, including all dihedral atom groups for molecule of current project
 
        *df_save_dir*
-           path to parent directory to create
-           subdirectory for saving the .csv files
-           
+           optional, path to the location to save results :class:`pandas.DataFrame`
+
+       *resname*
+           `resname` for the molecule as defined in 
+           the topology and trajectory
+
+       *molname*
+           molecule name to be used for labelling
+           plots, if different from `resname`
+
        :returns:
        
        *.csv.bz2*
-           Saves the results :class:`pandas.DataFrame` as a compressed
-           csv file (bz2).
+           saves the results :class:`pandas.DataFrame` as a compressed
+           csv file, default: .bz2
```

Comment:
The function does not return anything, therefore, remove the `:returns:`. We do not have a formal item to describe products of functions.

Instead, describe in the text body above what the output is, just using the sentence here.